docs/acceptance-criteria/uacs-rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,467 @@
+  group_size ) + 1` should become eligible (previous operator remains eligible)
+  - next members should become eligible after each fixed amount of blocks.
+
+| Relay request - relay entry submission
+| Upon receiving the transaction submitting the relay entry, the random beacon
+  contract should determine validity of the transaction.
+
+| Relay request - relay entry submission
+| When determining validity of the relay entry submission, the following aspects
+  should be taken into account: +
+  - is the submitter eligible to submit? +
+  - hasn’t the hard timeout for the relay request be reached?
+
+| Relay request - relay entry submission
+| When requesting a relay entry, it should be possible to pass an optional
+  address parameter - this is the address of a contract implementing
+  `IRandomBeaconConsumer` interface.
+
+| Relay request - relay entry submission
+| If the transaction submitting the new relay entry gets approved, then: +
+  - the BLS threshold signature should be accepted as a new relay entry +
+// TODO: revisit if 2 weeks removal time should be a governable parameter and if
+// it should be equall to 2 weeks or a different value
+  - operators who became eligible for submitting the relay entry before the
+  member who submitted the relay entry should be punished by being removed from
+  the sortition pool for a fixed amount of time (2 weeks) (in the same tx) +
+  - if an optional address of the `IRandomBeaconConsumer` interface was provided
+  in the tx, the interface should be called +
+  - the creation of a new signing group should be triggered (only sometimes, if
+  it’s the right time).
+
+| Relay request - relay entry submission
+| The transaction submitting relay entry is not reimbursable (i.e. the submitter
+  must provide the fee required for its processing).
+
+| Relay request - relay entry submission
+| The transaction submitting relay entry should have a predictable cost and the
+  gas cost of this transaction should be as low as possible, below 200k gas.
+
+| Relay request - relay entry submission
+| It should be possible to send low gas callbacks (up to the fixed gas cost
+  value) in the transaction submitting the relay entry.
+
+| Relay request - relay entry submission
+| Failure in the callback function should not revert the relay entry transaction.
+
+| Relay request - relay entry submission
+| The operation removing members from the sortition pool should be as cheap as
+  possible so that the member submitting relay entry does not have to pay any
+  additional costs for removing inactive members from the pool.
+
+| Relay request - relay entry timeouts
+| The soft relay entry timeout equals the group size multiplied by the number of
+  blocks for a member to become eligible to submit relay entry. The timer starts
+  at the moment when the first member becomes eligible (the moment when the
+  relay entry was requested). +
+  The hard relay entry timeout is a set value (measured in time units).
```

Comment:
Bear in mind the hard timeout timer starts at the moment when soft timeout is reached. The default value is `24h` but we use blocks to represent it in code. We assume `1 block = 15s`.

Reply:
Clarified, see 953ccbb65131b597e4fe3bba5ac4dca810aadcf1.